irs-api/src/main/java/org/eclipse/tractusx/irs/aaswrapper/job/CallbackResponderEventListener.java

```diff
@@ -60,87 +63,120 @@ class CallbackResponderEventListener {
     public static final String INVALID_CALLBACK_URL = "Invalid callback url '{}'.";
     private final UrlValidator urlValidator;
     private final RestTemplate restTemplate;
+    private final Map<String, Long> completedCallbacks;
 
     /* package */ CallbackResponderEventListener(
             @Qualifier(NO_ERROR_REST_TEMPLATE) final RestTemplate noErrorRestTemplate) {
         this.urlValidator = new UrlValidator(UrlValidator.ALLOW_LOCAL_URLS);
         this.restTemplate = noErrorRestTemplate;
+        this.completedCallbacks = new HashMap<>();
     }
 
     @Async
     @EventListener
     public void handleJobProcessingFinishedEvent(final JobProcessingFinishedEvent jobProcessingFinishedEvent) {
-        if (thereIsCallbackUrlRegistered(jobProcessingFinishedEvent.callbackUrl())) {
-            log.info("Processing of job has finished - attempting to notify job requestor");
-
-            final URI callbackUri = buildCallbackUri(jobProcessingFinishedEvent.callbackUrl(),
-                    jobProcessingFinishedEvent.jobId(), JobState.valueOf(jobProcessingFinishedEvent.jobState()));
-            if (urlValidator.isValid(callbackUri.toString())) {
-                log.info("Got callback url {} for jobId {} with state {}", callbackUri,
-                        jobProcessingFinishedEvent.jobId(), jobProcessingFinishedEvent.jobState());
-
-                try {
-                    final ResponseEntity<Void> callbackResponse = restTemplate.getForEntity(callbackUri, Void.class);
-                    log.info("Callback url pinged, received http status: {}, jobId {}", callbackResponse.getStatusCode(), jobProcessingFinishedEvent.jobId());
-                } catch (final ResourceAccessException resourceAccessException) {
-                    log.warn("Callback url is not reachable - connection timed out, jobId {}", jobProcessingFinishedEvent.jobId());
-                }
-            } else {
-                log.warn(INVALID_CALLBACK_URL, callbackUri);
-            }
+        if (StringUtils.isBlank(jobProcessingFinishedEvent.callbackUrl())) {
+            return;
+        }
+        log.info("Processing of job has finished - attempting to notify job requestor");
+
+        final URI callbackUri = buildCallbackUri(jobProcessingFinishedEvent.callbackUrl(),
+                jobProcessingFinishedEvent.jobId(), JobState.valueOf(jobProcessingFinishedEvent.jobState()));
+
+        if (!urlValidator.isValid(callbackUri.toString())) {
+            log.warn(INVALID_CALLBACK_URL, callbackUri);
+            return;
         }
```

Comment:
Is this also checked when creating a job? And if so, is the double check necessary?

Reply:
The callback URL is only validated for right protocol at job creation. 
Perhaps it would make more sense to validate the URL at job creation instead of when the callback is executed but that goes beyond the scope of this bug